src/content/articles/kubernetes-for-scala-developers-part-one/index.mdx

```diff
@@ -0,0 +1,1417 @@
+---
+author: gedeon-tabela
+category: guide
+difficulty: intermediate
+excerpt: Learn how to deploy and scale your Scala application using Kubernetes
+publishedDate: 2025-03-10
+tags: [kubernetes, scala, devops, scalability]
+title: "Scala applications containers Orchestration with Kubernetes(Part 1)"
+updatedDate: 2025-03-11
+---
+
+
+## Introduction: Why Kubernetes Matters for Scala Developers
+Imagine that you've built an interesting Scala application, and you are ready to deploy it for your users to consume
+your services. As a good developer you foresee numerous real challenges to address for your application to efficiently
+serve the public. How will the application scale up and down in response to the traffic, avoid service downtime, maximize
+cost-efficiency? The most efficient solution for your worries is Kubernetes.
+
+If you as a Scala developer you are struggling with your infrastructure, auto-scaling and some deployment automation
+headaches, this first part guide will guide you through building a powerful application ecosystem by introducing you to
+the core concepts of a Kubernetes cluster. By the end of this first part you'll be able to understand the architecture
+of a Kubernetes cluster, the main components of Kubernetes, how to scale up or down your application, how to apply
+zero-downtime deployments and more. The second part will guide you through continuous integration and deployment, monitoring,
+and help you build a production-ready Kubernetes cluster.
+
+## Kubernetes: Your Application's New Best Friend
+Kubernetes is not just another helper tool for deploying applications. It's a complete infrastructural ecosystem positioning
+itself as the most modern solution for deploying resilient and elastics applications. It is a hyper-intelligent and extensible
+deployment orchestrator that manages all aspects of your application's lifecycle.
+
+In this article we'll deploy a Scala cars parking management application built with Play Framework uses an in-memory database
+for simplicity.
+The Docker images that we'll be using are:
+`pfpw:20.20.31` and `pfpw:20.20.32` publicly available on Docker Hub. It exposes two api's that we can test:
```

Comment:
Don't just include the Docker Hub tags. Include the Scala code as well as the `build.sbt` settings you used to dockerize the app.